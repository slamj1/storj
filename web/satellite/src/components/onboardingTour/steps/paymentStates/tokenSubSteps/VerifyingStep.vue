// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template src="./verifyingStep.html"></template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import BackImage from '@/../static/images/account/billing/back.svg';
import VerifyingImage from '@/../static/images/onboardingTour/verifying.svg';

@Component({
    components: {
        VerifyingImage,
        BackImage,
    },
})

export default class VerifyingStep extends Vue {
    /**
     * Holds logic for button click.
     * Sets area to default state.
     */
    public onBackClick(): void {
        this.$emit('setDefaultState');
    }
}
</script>

<style scoped lang="scss" src="./verifyingStep.scss"></style>
