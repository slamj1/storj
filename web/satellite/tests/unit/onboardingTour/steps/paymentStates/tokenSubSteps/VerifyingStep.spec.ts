// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import VerifyingStep from '@/components/onboardingTour/steps/paymentStates/tokenSubSteps/VerifyingStep.vue';

import { shallowMount } from '@vue/test-utils';

describe('VerifyingStep.vue', () => {
    it('renders correctly', async (): Promise<void> => {
        const wrapper = shallowMount(VerifyingStep);

        expect(wrapper).toMatchSnapshot();

        await wrapper.find('.verifying-step__back-button').trigger('click');

        expect(wrapper.emitted()).toEqual({ 'setDefaultState': [[]] });
    });
});
