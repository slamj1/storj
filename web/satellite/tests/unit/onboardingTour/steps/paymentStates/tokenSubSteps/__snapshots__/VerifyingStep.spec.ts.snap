// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyingStep.vue renders correctly 1`] = `
<div class="verifying-step">
  <verifyingimage-stub></verifyingimage-stub>
  <h2 class="verifying-step__title">Verifying Payment</h2> <span class="verifying-step__sub-title">
        Refresh this page in 10-15 minutes to continue after your purchase is verified. In the meantime, see how
        easy it is to get started visiting documentation page, or join the community to get involved.
    </span>
  <div class="verifying-step__buttons-area"><a href="https://documentation.storj.io" target="_blank" class="verifying-step__buttons-area__how-to-button">
      Documentation
    </a> <a href="https://storj.io/community/" target="_blank" class="verifying-step__buttons-area__community-button">
      Community
    </a></div>
  <div class="verifying-step__back-button">
    <backimage-stub></backimage-stub>
  </div>
</div>
`;
